server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexAliasesService.java

```diff
@@ -69,17 +71,9 @@ public MetadataIndexAliasesService(
     }
 
     public void indicesAliases(final IndicesAliasesClusterStateUpdateRequest request, final ActionListener<AcknowledgedResponse> listener) {
-        submitUnbatchedTask("index-aliases", new AckedClusterStateUpdateTask(Priority.URGENT, request, listener) {
```

Comment:
Previously the task was an `AckedClusterStateUpdateTask` and I think we want to keep that behaviour. That means using `success(ClusterStateAckListener clusterStateAckListener)` instead of `success(Runnable onPublicationSuccess)`, which will give you a place to implement the missing plumbing.

Reply:
Thank you for the tip, it took me a bit but I think now I understand how they connect and what I was missing with the previous implementation.